classes/invitation/core/Invitation.php

```diff
@@ -54,53 +94,105 @@ public function __construct(?InvitationModel $invitationModel = null)
         $this->fillFromPayload();
     }
 
-    public function initialize(?int $userId = null, ?int $contextId = null, ?string $email = null)
+    /**
+     * Function that adds the invitation to the database, initialising its main attributes.
+     * Either userId or email is taken into account - if both are defined, the userId is passed into the invitation.
+     * It removes all other InvitationStatus::INITIALIZED invitation from the database, that correspont to
+     * the same main attributes.
+     */
+    public function initialize(?int $userId = null, ?int $contextId = null, ?string $email = null, ?int $inviterId = null): void
     {
         if (!isset($userId) && !isset($email)) {
-            throw new Exception("Invitation should contain at least one user id or an invited email')");
+            throw new Exception("Invitation should contain the user id or an invited email.')");
+        }
+
+        $userIdUsed = null;
+        $emailUsed = null;
+        if (isset($userId)) {
+            $userIdUsed = $userId;
+        } else {
+            $emailUsed = $email;
         }
 
-        $this->invitationModel->userId = $userId;
+        InvitationModel::byStatus(InvitationStatus::INITIALIZED)
+            ->when($userIdUsed !== null, function ($query) use ($userIdUsed) {
+                return $query->byUserId($userIdUsed);
+            })
+            ->when($contextId !== null, function ($query) use ($contextId) {
+                return $query->byContextId($contextId);
+            })
+            ->when($emailUsed !== null, function ($query) use ($emailUsed) {
+                return $query->byEmail($emailUsed);
+            })
+            ->byType($this->getType())
+            ->delete();
+
+        $this->invitationModel->userId = $userIdUsed;
         $this->invitationModel->contextId = $contextId;
-        $this->invitationModel->email = $email;
+        $this->invitationModel->email = $emailUsed;
+        $this->invitationModel->inviterId = $inviterId;
 
         $this->invitationModel->status = InvitationStatus::INITIALIZED;
 
         $this->invitationModel->save();
     }
 
-    public function fillFromArgs(array $args)
+    /**
+     * Used to fill the invitation's properties from the model's payload values.
+     * if the $propertyType has values in, then the property is filled by the 
+     * fromArray function of the given Object correlation
+     */
+    protected function fillFromPayload(): void
+    {
+        if ($this->invitationModel->payload) {
+            foreach ($this->invitationModel->payload as $key => $value) {
+                if (property_exists($this, $key)) {
```

Comment:
Are there cases where something will be in the payload and *shouldn't* be here as a property? If not, I wonder whether it wouldn't be better to...
```php
if (!property_exists($this, $key)) {
    error_log("Payload property {$key} does not exist on object " . get_class($this));
    continue;
}
```
...but see my comment about payload data mixing into this class first.